File: SRC/frontend/src/services/adminService.js
/**
 * Admin Service
 * Quan ly nguoi dung, duyet nha hang, thong ke
 */
import api from '@/services/api'
import ownerRequestService from '@/services/ownerRequestService'
import shipperRequestService from '@/services/shipperRequestService'

export default {
  // --- User Management ---
  async getAllUsers() {
    return api.get('/api/admin/users')
  },

  async toggleUserActive(userId, isActive) {
    return api.patch(`/api/admin/users/${userId}/status`, { active: isActive })
  },

  async updateUserRole(userId, role) {
    return api.patch(`/api/admin/users/${userId}/role`, { role })
  },

  async deleteUser(userId) {
    return api.delete(`/api/admin/users/${userId}`)
  },

  // --- Restaurant Approval ---
  async getPendingRestaurants() {
    return api.get('/api/admin/restaurants/pending')
  },

  async approveRestaurant(restaurantId) {
    return api.post(`/api/admin/restaurants/${restaurantId}/approve`, {
      approved: true,
      note: null,
    })
  },

  async rejectRestaurant(restaurantId, reason) {
    return api.post(`/api/admin/restaurants/${restaurantId}/approve`, {
      approved: false,
      note: reason || null,
    })
  },

  // --- Stats ---
  async getSystemStats() {
    return api.get('/api/admin/stats')
  },

  // --- Owner Request Approval ---
  async getPendingOwnerRequests() {
    return ownerRequestService.getPendingRequests()
  },

  async approveOwnerRequest(requestId) {
    return ownerRequestService.processRequest(requestId, true, null)
  },

  async rejectOwnerRequest(requestId, reason) {
    return ownerRequestService.processRequest(requestId, false, reason || null)
  },

  // --- Shipper Request Approval ---
  async getPendingShipperRequests() {
    return shipperRequestService.getPendingRequests()
  },

  async approveShipperRequest(requestId) {
    return shipperRequestService.processRequest(requestId, true, null)
  },

  async rejectShipperRequest(requestId, reason) {
    return shipperRequestService.processRequest(requestId, false, reason || null)
  },
}

File: SRC/backend/src/main/java/com/example/server/service/AdminService.java
package com.example.server.service;

import com.example.server.dto.report.AdminStatsResponse;
import com.example.server.dto.restaurant.RestaurantApprovalRequest;
import com.example.server.dto.restaurant.RestaurantCardResponse;
import com.example.server.dto.user.UserProfileResponse;
import com.example.server.dto.user.UserRoleUpdateRequest;
import com.example.server.dto.user.UserStatusUpdateRequest;
import java.util.List;

public interface AdminService {
    void approveRestaurant(Long restaurantId, RestaurantApprovalRequest request);
    void updateUserRole(Long userId, UserRoleUpdateRequest request);
    void updateUserStatus(Long userId, UserStatusUpdateRequest request);
    List<UserProfileResponse> getAllUsers();
    List<RestaurantCardResponse> getPendingRestaurants();
    AdminStatsResponse getSystemStats();
}
